scipy/stats/tests/test_qmc.py

```diff
@@ -534,23 +535,43 @@ def test_sample(self, *args):
         pytest.skip("Not applicable: the value of reference sample is"
                     " implementation dependent.")
 
+    @pytest.mark.parametrize("orthogonal", [False, True])
+    @pytest.mark.parametrize("centered", [False, True])
     @pytest.mark.parametrize("optimization", [None, "random-CD"])
-    def test_sample_stratified(self, optimization):
-        d, n = 4, 20
+    def test_sample_stratified(self, optimization, centered,
+                               orthogonal):
+        seed = np.random.RandomState(123456)
+        d, n = 4, 25
         expected1d = (np.arange(n) + 0.5) / n
         expected = np.broadcast_to(expected1d, (d, n)).T
 
-        engine = self.engine(d=d, scramble=False, centered=True,
-                             optimization=optimization)
+        engine = qmc.LatinHypercube(d=d, centered=centered,
+                                    orthogonal=orthogonal,
+                                    optimization=optimization,
+                                    seed=seed)
         sample = engine.random(n=n)
+        assert sample.shape == (n, d)
+        assert engine.num_generated == n
+
         sorted_sample = np.sort(sample, axis=0)
 
-        assert_equal(sorted_sample, expected)
         assert np.any(sample != expected)
-
         assert_allclose(sorted_sample, expected, atol=0.5 / n)
         assert np.any(sample - expected > 0.5 / n)
 
+    @pytest.mark.parametrize("centered", [False, True])
+    def test_orthogonal_array(self, centered):
+        seed = np.random.RandomState(123456)
+        p = 5
+
+        engine = qmc.LatinHypercube(d=6, centered=centered,
+                                    orthogonal=True, seed=seed)
+        oa_lhs_sample = engine.random(p**2)
+        for i, j in combinations(range(engine.d), 2):
+            samples_2d = oa_lhs_sample[:, [i, j]]
+            assert np.all(np.unique((samples_2d * p).astype(int),
+                                    return_counts=True, axis=0)[1] == 1)
```

Comment:
I think this does what it's intended to do if we know that the values in `oa_lhs_sample` are initially between 0 and 1. If that were not true, then we are just making sure that the rows are unique after the transformation, which is not so hard to achieve if the original values can be of any magnitude.

For instance, suppose instead of 
```python3
oa_lhs_sample = engine.random(p**2)
```
we had
```
oa_lhs_sample = np.random.rand(p**2, 6) * 100
```
Then the test would still pass.

So in conjunction with all the other tests, this test is fine, but I could be stronger on its own, e.g.:

```python3
unique_elements = [0, 1, 2, 3, 4]
desired = list(product(unique_elements, unique_elements))
        
for i, j in combinations(range(engine.d), 2):
    samples_2d = oa_lhs_sample[:, [j, i]]
    res  = (samples_2d * p).astype(int)
    assert_equal(res, desired)
```
so that it's more explicit that the values are those that we expect.

Please also consider adding this to `test_sample_stratified`, executing only `if orthogonal:`. That way it's explicit that the same array passes _both_ tests. Alternatively, use the same `d` (and `n` and `seed`) in both tests, so that the two tests are performed on the same array.

Reply:
Ok thanks, I will use this.

Reply:
It's interesting because I did not saw this due to the fact that with a plain LHS, it will fail because it's of strength 1 and not 2.

Reply:
I'm sorry, I don't understand. If the strength is 2, the second test should run and pass the new test. If the strength is only 1, it only needs to pass the existing test. So if you put them in the same `test_sample_stratified`, there would need to be a branching `if` statement to conditionally run the new test based on the parameters passed into `LatinHypercube` (i.e. `orthogonal` or `strength`, whatever we end up with).

Reply:
What I mean is that the code I had was not passing on a LHS but passed on the random and the OA. And your version is only passing for OA.